File: test-junit5/src/test/java/io/micronaut/test/junit5/server/ProcessServerTest.java
package io.micronaut.test.junit5.server;

import io.micronaut.context.annotation.Property;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpStatus;
import io.micronaut.http.client.RxHttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.runtime.server.EmbeddedServer;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import io.micronaut.test.support.server.TestExecutableEmbeddedServer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import jakarta.inject.Inject;
import java.util.Collections;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;


@MicronautTest
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
// tag::executable[]
@Property(
    name = TestExecutableEmbeddedServer.PROPERTY,
    value = "src/test/apps/test-app.jar"
)
// end::executable[]
public class ProcessServerTest implements TestPropertyProvider {

    @Inject
    EmbeddedServer embeddedServer;


    @Inject
    @Client("/")
    RxHttpClient client;

    @Test
    void testServerAvailable() {
        HttpResponse<String> response = client.exchange("/test", String.class).blockingFirst();

        assertTrue(
                embeddedServer instanceof TestExecutableEmbeddedServer
        );
        assertTrue(
                embeddedServer.isRunning()
        );
        assertEquals(
                HttpStatus.OK,
                response.status()
        );
        assertEquals(
                "Result = good",
                response.body()
        );
    }

    @Override
    public Map<String, String> getProperties() {
        return Collections.singletonMap(
                "test.property", "good"
        );
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/server/ProcessServerTest.py
from typing import Annotated

import java
from jakarta.inject import Inject
from java.lang import String
from micronaut.context.annotation import Property
from micronaut.http import HttpStatus
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.runtime.server import EmbeddedServer
from micronaut.test.extensions.junit5.annotation import MicronautTest
from micronaut.test.support.server import TestExecutableEmbeddedServer
from org.junit.jupiter.api import Test


@MicronautTest
# tag::executable[]
@Property(
    name="micronaut.test.server.executable",
    value="../test-junit5/src/test/apps/test-app.jar"
)
# end::executable[]
@Property(name="test.property", value="good")
class ProcessServerTest:

    embedded_server: Annotated[EmbeddedServer, Inject]

    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def test_server_available(self):
        response = self.client.toBlocking().exchange("/test", String)

        assert java.instanceof(self.embedded_server, TestExecutableEmbeddedServer)
        assert self.embedded_server.isRunning()
        assert HttpStatus.OK == response.status()
        assert "Result = good" == response.body()
